### PR TITLE
docs: Move CONTRIBUTING.md to development guide and document backward…

### PR DESCRIPTION
**Pull request by achals**
…s compatibility policy

Signed-off-by: Achal Shah <achals@gmail.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: docs/project/development-guide.md [hunk 1]
```diff
@@ -1,72 +1,436 @@
-# Development guide
+# Development Guide: Main Feast Repository<
```
Comment (felixwang9817) [inline docs/project/development-guide.md:1]: stray <

File: docs/project/development-guide.md [hunk 2]
```diff
@@ -1,72 +1,436 @@
...
+- [Feast Python SDK / CLI](#feast-python-sdk--cli)
+- [Feast Java Serving](#feast-java-serving)
+- [Feast Go Client](#feast-go-client)
 
-* [Project Structure](development-guide.md#repository-structure)
-* [Making a Pull Request](development-guide.md#making-a-pull-request)
-* [Feast Data Storage Format](development-guide.md#feast-data-storage-format)
-* [Feast Protobuf API](development-guide.md#feast-protobuf-api)
-* [Maintainer Guide](./maintainers.md)
+Please see [this page](https://docs.feast.dev/reference/codebase-structure) for more details on the structure of the entire codebase.
```
Comment (felixwang9817) [inline docs/project/development-guide.md:48]: nit: use a relative path for consistency

File: docs/project/development-guide.md [hunk 3]
```diff
@@ -1,72 +1,436 @@
...
 
-> Learn How the Feast [Contributing Process](contributing.md) works.
+## Compatibility
 
-## Making a Pull Request
+The Feast project
```
Comment (felixwang9817) [inline docs/project/development-guide.md:52]: stray or something needs to be added here

File: docs/project/development-guide.md [hunk 4]
```diff
@@ -1,72 +1,436 @@
...
+## Compatibility
 
-## Making a Pull Request
+The Feast project 
 
-{% hint style="info" %}
-See also the CONTRIBUTING.md in the corresponding GitHub repository \(e.g. [main repo doc](https://github.com/feast-dev/feast/blob/master/CONTRIBUTING.md)\)
-{% endhint %}
+## Community
+See [Contribution process](https://docs.feast.dev/project/contributing) and [Community](https://docs.feast.dev/community) for details on how to get more involved in the community.
```
Comment (felixwang9817) [inline docs/project/development-guide.md:55]: also relative paths for consistency

File: docs/project/development-guide.md [hunk 5]
```diff
@@ -1,72 +1,436 @@
...
+- [Feast Dev Mailing List](https://groups.google.com/g/feast-dev)
+- [Community Calendar](https://calendar.google.com/calendar/u/0?cid=ZTFsZHVhdGM3MDU3YTJucTBwMzNqNW5rajBAZ3JvdXAuY2FsZW5kYXIuZ29vZ2xlLmNvbQ)
+  - Includes biweekly community calls at 10AM PST
 
-Our preference is the use of `git rebase` instead of `git merge` : `git pull -r`
+## Making a pull request
+We use the convention that the assignee of a PR is the person with the next action.
 
-### Signing commits
+This means that often, the assignee may be empty (if no reviewer has been found yet), the reviewer, or the PR writer if there are comments to be addressed.
```
Comment (felixwang9817) [inline docs/project/development-guide.md:67]: sentence doesn't quite make sense to me

File: docs/project/development-guide.md [hunk 6]
```diff
@@ -1,72 +1,436 @@
...
+7. Export GCP specific environment variables in your workflow. Namely,
+  ```sh
+  export GCS_REGION='[your gcs region e.g US]'
+  export GCS_STAGING_LOCATION='[your gcs staging location]'
+  ```
+**NOTE**: Your `GCS_STAGING_LOCATION` should be in the form `gs://<bucket name>` where the bucket name is from step 2.
+
+8. Once authenticated, you should be able to run the integration tests for BigQuery without any failures.
 
-## Feast Protobuf API
```
Comment (felixwang9817) [inline docs/project/development-guide.md:58]: this seems to have been completed deleted; was that intentional?

File: docs/project/development-guide.md (hunk 4 above)
Comment (achals) [reply docs/project/development-guide.md:55]: done

File: docs/project/development-guide.md (hunk 3 above)
Comment (achals) [reply docs/project/development-guide.md:52]: fixed

File: docs/project/development-guide.md (hunk 2 above)
Comment (achals) [reply docs/project/development-guide.md:48]: done

File: docs/project/development-guide.md (hunk 1 above)
Comment (achals) [reply docs/project/development-guide.md:1]: oops

File: docs/project/development-guide.md (hunk 5 above)
Comment (achals) [reply docs/project/development-guide.md:67]: Came in from the CONTRIBUTING file but I'll fix

File: docs/project/development-guide.md (hunk 6 above)
Comment (achals) [reply docs/project/development-guide.md:58]: Yea I souped up the protobuf section elsewhere in this doc but this snipped wasn't super useful.

**Pull request merged**